### PR TITLE
PMM-6272 New settings page

### PR DESCRIPTION
**Pull request by tiagomotasantos**
[PMM-6272](https://jira.percona.com/browse/PMM-6272)

- [ ] https://github.com/Percona-Lab/pmm-submodules/pull/976

### CODE REVIEW

File: pmm-app/src/pmm-settings/components/Advanced/SwitchRow.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react';
+import { Switch, useTheme } from '@grafana/ui';
+import { getSettingsStyles } from 'pmm-settings/Settings.styles';
+import { LinkTooltip } from 'shared/components/Elements/LinkTooltip/LinkTooltip';
+import { getStyles } from './Advanced.styles';
+
+
+export interface SwitchRowProps {
```
Comment (lunaticusgreen) [inline pmm-app/src/pmm-settings/components/Advanced/SwitchRow.tsx:8]: Please move types to separate file named SwitchRow.types.ts and do it for all interfaces

File: pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  updateSettings: (body: any, callback: LoadingCallback) => void;
+}
+
+export const MetricsResolution: FC<MetricsResolutionProps> = ({ metricsResolutions, updateSettings }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  const settingsStyles = getSettingsStyles(theme);
+  const initialResolutions = removeUnits(metricsResolutions);
+  const [resolution, setResolution] = useState(getResolutionValue(metricsResolutions).key);
+  const [updatedResolutions, updateResolutions] = useState(initialResolutions);
```
Comment (lunaticusgreen) [inline pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx:26]: We can use MetricsResolutions type here

File: pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        name="resolutions"
+        dataQa="metrics-resolution-radio-button-group"
+        className={styles.resolutionsRadioButtonGroup}
+        onChange={(key) => setResolution(key)}
+      />
+      <input
+        type="number"
+        className={styles.resolutionInput}
+        value={updatedResolutions.lr}
+        disabled={resolution !== 'custom'}
```
Comment (lunaticusgreen) [inline pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx:67]: Let's move this string to type,  we use it may times

Comment (lunaticusgreen) [reply pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx:67]: And 'lr', 'mr', 'hr' too

File: pmm-app/src/pmm-settings/Settings.service.ts [hunk 4]
```diff
@@ -1,10 +1,52 @@
...
+export type LoadingCallback = (value: boolean) => void;
 
 export const SettingsService = {
-  getSettings() {
-    return apiRequest.post<any, any>('/v1/Settings/Get', {});
+  async getSettings(callback: LoadingCallback) {
+    let response: any = null;
+
+    try {
+      callback(true);
```
Comment (atymchuk) [inline pmm-app/src/pmm-settings/Settings.service.ts:13]: Does it make sense to call the callback before the request, not after?

File: pmm-app/src/pmm-settings/Settings.service.ts [hunk 5]
```diff
@@ -1,10 +1,52 @@
...
+import { Messages } from './Settings.messages';
+import { Settings } from './Settings.types';
+
+export type LoadingCallback = (value: boolean) => void;
 
 export const SettingsService = {
-  getSettings() {
-    return apiRequest.post<any, any>('/v1/Settings/Get', {});
+  async getSettings(callback: LoadingCallback) {
+    let response: any = null;
```
Comment (atymchuk) [inline pmm-app/src/pmm-settings/Settings.service.ts:10]: No need to initialize.

File: pmm-app/src/pmm-settings/Settings.service.ts [hunk 6]
```diff
@@ -1,10 +1,52 @@
...
+
+    return response;
   },
-  setSettings(body) {
-    return apiRequest.post<any, any>('/v1/Settings/Change', body);
+  async setSettings(body, callback: LoadingCallback) {
+    let response: any = null;
+
+    try {
+      callback(true);
```
Comment (atymchuk) [inline pmm-app/src/pmm-settings/Settings.service.ts:28]: Callback should fire after the API.

File: pmm-app/src/pmm-settings/components/AlertManager/AlertManager.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      rulesTooltip,
+      urlLabel,
+      urlTooltip
+    },
+    tooltipLinkText
+  } = Messages;
+  const [url, setUrl] = useState(alertManagerUrl);
+  const [rules, setRules] = useState(alertManagerRules);
+  const [loading, setLoading] = useState(false);
+  const disableAction = () => (
```
Comment (atymchuk) [inline pmm-app/src/pmm-settings/components/AlertManager/AlertManager.tsx:38]: More semantic: isActionDisabled

File: pmm-app/src/pmm-settings/Settings.service.ts (hunk 4 above)
Comment (tiagomotasantos) [reply pmm-app/src/pmm-settings/Settings.service.ts:13]: The "callback" is the LoadingCallback, to set loading to true/false, so it needs to be both, before and after the request. But maybe I should call it "setLoading" then? To be easier to understand

File: pmm-app/src/pmm-settings/components/Advanced/Advanced.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                data-qa="advanced-retention-input"
+              />
+              <span className={styles.retentionUnitslabel}>{retentionUnits}</span>
+            </td>
+          </tr>
+          <SwitchRow
+            label={telemetryLabel}
+            tooltip={telemetryTooltip}
+            tooltipLinkText={tooltipLinkText}
+            link={`${GUI_DOC_URL}#server-admin-gui-telemetry`}
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply pmm-app/src/pmm-settings/components/Advanced/Advanced.tsx:98]: Moved!

File: pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx (hunk 3 above)
Comment (tiagomotasantos) [reply pmm-app/src/pmm-settings/components/MetricsResolution/MetricsResolution.tsx:68]: Done. Move the presets (rare, standard, frequent, custom) and the intervals (lr, mr, hr) to enums.

File: pmm-app/src/pmm-settings/Settings.styles.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import { stylesFactory } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { css } from 'emotion';
+
+export const getSettingsStyles = stylesFactory((theme: GrafanaTheme) => {
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply pmm-app/src/pmm-settings/Settings.styles.ts:5]: Looking good in light mode now 😁😁😁

File: pmm-app/package.json [hunk 10]
```diff
@@ -18,6 +18,7 @@
     "dev": "grafana-toolkit plugin:dev",
     "watch": "grafana-toolkit plugin:dev --watch",
     "e2e": "docker run --net=host -v $PWD:/tests -v $PWD/node_modules:/node_modules --env-file env.generated.list codeception/codeceptjs:2.6.1 codeceptjs run-multiple parallel --all --debug --steps -c pr.codecept.js --grep '(?=.*)^(?!.*@visual-test)^(?!.*@not-pr-pipeline)'",
+    "e2e:local": "docker run --net=host -v $PWD:/tests -v $PWD/node_modules:/node_modules codeception/codeceptjs:2.6.1 codeceptjs run-multiple parallel --all --debug --steps -c pr-local.codecept.js --grep '(?=.*)^(?!.*@visual-test)^(?!.*@not-pr-pipeline)'",
```
Comment (puneet0191) [inline pmm-app/package.json:21]: we removed this tag visual-test, it was replaced with `@not-ui-pipeline` could you please replace on `e2e` and `e2e:local` both

File: pmm-app/pr-local.codecept.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    REST: {
+      endpoint: process.env.PMM_UI_URL || 'http://localhost/'
+    },
+  },
+  include: {
+    pmmDemoPage: './tests/pages/pmmDemoPage.js',
+    homePage: './tests/pages/homePage.js',
+    remoteInstancesPage: './tests/pages/remoteInstancesPage.js',
+    adminPage: './tests/pages/adminPage.js',
+    qanPage: './tests/QAN/qanPage.js',
```
Comment (puneet0191) [inline pmm-app/pr-local.codecept.js:32]: looks like outdated.

File: pmm-app/package.json [hunk 12]
```diff
@@ -17,7 +17,8 @@
     "coverage": "grafana-toolkit plugin:test --coverage",
     "dev": "grafana-toolkit plugin:dev",
     "watch": "grafana-toolkit plugin:dev --watch",
-    "e2e": "docker run --net=host -v $PWD:/tests -v $PWD/node_modules:/node_modules --env-file env.generated.list codeception/codeceptjs:2.6.1 codeceptjs run-multiple parallel --all --debug --steps -c pr.codecept.js --grep '(?=.*)^(?!.*@visual-test)^(?!.*@not-pr-pipeline)'",
+    "e2e": "docker run --net=host -v $PWD:/tests -v $PWD/node_modules:/node_modules --env-file env.generated.list codeception/codeceptjs:2.6.1 codeceptjs run-multiple parallel --all --debug --steps -c pr.codecept.js --grep '(?=.*)^(?!.*@not-ui-pipeline)^(?!.*@not-pr-pipeline)'",
+    "e2e:local": "docker run --net=host -v $PWD:/tests -v $PWD/node_modules:/node_modules codeception/codeceptjs:2.6.1 codeceptjs run-multiple parallel --all --debug --steps -c pr-local.codecept.js --grep '(?=.*)^(?!.*@not-ui-pipeline)^(?!.*@not-pr-pipeline)'",
```
Comment (puneet0191) [inline pmm-app/package.json:21]: why 2 times same tag?

**Pull request merged**